### ISSUE TITLE
gem dependencies and bundler not picking up groups

### ISSUE THREAD

**Issue by jondot** [colinsurprenant/redstorm#49]
Hey guys,

I've traced a problem here https://github.com/colinsurprenant/redstorm/blob/master/lib/tasks/red_storm.rake#L160

Which causes `redstorm bundle` to not copy any gems.

Where bundler keeps the default group as the symbol `:default` and not the string `'default'`:

```
irb(main):019:0> Bundler.definition.groups
=> [:default]
```

This is bundler 1.1.4, I didn't make a pull request because I'm not sure what it is the string `'default'` in the first place... any idea?

thanks

**Comment by jondot** [comment]
Nevermind the previous theory. Looks like it has to do with groups being blank (`""`) and not nil.
Following snippet fixes it for me:

``` ruby
  require 'bundler'
  # get rid of rake magic hash
  args = args.to_hash
  # normalize manually
  args[:groups] = 'default' if (args[:groups].nil? || args[:groups]== "")
  # continue as usual..
  groups = args[:groups].split(':').map(&:to_sym)
  Bundler.definition.specs_for(groups).each do |spec|
```

makes sense?

**Comment by fractaloop** [comment]
From what I can tell, you'll want to put the gems required from your topology in a special group and then bundle that.

Your Gemfile would look something like:

``` ruby
gem 'rake'

group :topology do
  gem 'json'
end
```

And bundle them with `redstorm bundle topology`

**Comment by fractaloop** [comment]
For more background on the Bundler SNAFU check out https://github.com/colinsurprenant/redstorm/pull/33

**Comment by colinsurprenant** [comment]
@fractaloop is right in that you should ideally put the topology gems into a group and reference that group when doing the `redstorm bundle GROUP_NAME`, this is also what I suggested in the README. 

BUT :) `redstorm bundle` without parameter should correctly install gems not in any group. There is in fact a bug with the "default" group handling - thanks for spotting this. I am about to commit the fix for release 0.6.3.

**Comment by colinsurprenant** [comment]
And btw @jondot the way I personally setup a RedStorm project is by first creating a Gemfile with 

``` ruby
source :rubygems
gem "redstorm"
group :topology do
  gem "redis"
end
```

then I do a `bundle install`, `redstorm install` and for the topology gems I do a `redstorm bundle topology` and so on.

Make sense?
